### PR TITLE
Upgrade Go to v1.21

### PR DESCRIPTION
**Pull request by mcbenjemaa**

### CODE REVIEW

**Comment by avorima** [comment]
Tests are failing because you mix `context.TODO()` and `context.Background()` in your mock assertions

**Comment by mcbenjemaa** [comment]
Yeah, i know. 

In 1.20 It ignores that.

**Comment by wikkyk** [comment]
ref #116

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,9 +9,6 @@ on:
 permissions:
   contents: write # Allow to create a release.
 
-env:
-  go_version: 1.20
```
Comment (wikkyk) [inline .github/workflows/release.yml:13]: shouldn't this stay in, changed to 1.21?

**Comment by mcbenjemaa** [comment]
Please run e2e tests

**Pull request merged**